core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -70,50 +69,43 @@ public void start() throws Exception {
     }
 
     while (!Thread.interrupted()) {
-      if (mServingThread == null) {
-        // We are in secondary mode. Nothing to do until we become the primary.
-        mLeaderSelector.waitForState(State.PRIMARY);
-        LOG.info("Transitioning from secondary to primary");
-        AbstractJournalSystem.ALLOW_JOURNAL_MODIFY.set(true);
-        mJournalSystem.setMode(Mode.PRIMARY);
-        stopMasters();
-        LOG.info("Secondary stopped");
-        startMasters(true);
-        mServingThread = new Thread(() -> {
-          try {
-            startServing(" (gained leadership)", " (lost leadership)");
-          } catch (Throwable t) {
+      startMasters(false);
```

Comment:
before we didn't actually start the secondary masters unless the master became primary and then became secondary again. Secondary masters don't actually do anything though, so this was ok.